pennylane/beta/queuing/operation.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# pylint: disable=protected-access
+r"""
+This module contains the abstract base classes for defining PennyLane
+operations and observables.
+"""
+
+import pennylane as qml
+
+# --------------------
+# Beta related imports
+# --------------------
+
+from pennylane.operation import Tensor
+
+class BetaTensor(Tensor):
+    """Container class representing tensor products of observables.
+
+    To create a tensor, simply initiate it like so:
+
+    >>> T = Tensor(qml.PauliX(0), qml.Hermitian(A, [1, 2]))
+
+    You can also create a tensor from other Tensors:
+
+    >>> T = Tensor(T, qml.PauliZ(4))
+
+    The ``@`` symbol can be used as a tensor product operation:
+
+    >>> T = qml.PauliX(0) @ qml.Hadamard(2)
+    """
+
+    # pylint: disable=abstract-method,too-few-public-methods
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.queue()
+
+    def queue(self):
+        """Queues the Tensor instance and updates the ownership related info if applicable."""
+        qml.QueuingContext.append(self, owns=tuple(self.obs))
+
+        if qml.QueuingContext.active_context().__class__.__name__ == "AnnotatedQueue":
+            # If the QueuingContext.update_info is not redefined by the
+            # specific queue type, an infinite recursion arises
+            # Hence updating only for an AnnotatedQueue
```

Comment:
Looked into this a bit more:
* `QueuingContext`: defines `update_info`, but it also calls the `update_info` method of the `active_context`
     * `Queue`: doesn't define `update_info` method, will still use the `QueuingContext` `update_info` method if it is an `active_context` --- resulting in infinite recursion
     * `AnnotatedQueue`: defines `update_info`, therefore if an `AnnotatedQueue` instance is the `active_context` then we don't have infinite recursion (intended use)

This solution supports both `qml.QueuingContext.update_info` and `qml._queuing.AnnotatedQueue().update_info` calls.

Reply:
I think `QueuingContext.update_info` snuck in during the last PR (don't recall it being part of the proposal), so if it's causing issues, it can be made a dedicated method of `AnnotatedQueue`

Reply:
Thank you! Changed it such that now `QueuingContext.update_info` will depend on a private `_update_info` instance method which by default raises a `NotImplementedError`.